spring-pulsar-spring-cloud-stream-binder/src/main/java/org/springframework/pulsar/spring/cloud/stream/binder/provisioning/PulsarTopicProvisioner.java

```diff
@@ -34,74 +36,63 @@
 public class PulsarTopicProvisioner implements
 		ProvisioningProvider<ExtendedConsumerProperties<PulsarConsumerProperties>, ExtendedProducerProperties<PulsarProducerProperties>> {
 
-	// TODO: Retrieve partitions through config
+	private final PulsarAdministration pulsarAdministration;
+
+	private final PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties;
+
+	public PulsarTopicProvisioner(PulsarAdministration pulsarAdministration,
+			PulsarBinderConfigurationProperties pulsarBinderConfigurationProperties) {
+		this.pulsarAdministration = pulsarAdministration;
+		this.pulsarBinderConfigurationProperties = pulsarBinderConfigurationProperties;
+	}
+
 	@Override
 	public ProducerDestination provisionProducerDestination(String name,
 			ExtendedProducerProperties<PulsarProducerProperties> properties) throws ProvisioningException {
-		PulsarTopic pulsarTopic = PulsarTopic.builder(name).numberOfPartitions(1).build();
-		return new PulsarProducerDestination(pulsarTopic.topicName(), pulsarTopic.numberOfPartitions());
+
+		int partitionCount = this.pulsarBinderConfigurationProperties.partitionCount();
+		int partitionCountOnBinding = properties.getPartitionCount(); // leverage the
+																		// partition count
+																		// from core
+		if (partitionCountOnBinding > 1) {
```

Comment:
Maybe we can make the default for the binding props PC `Integer null` and then validate when set > 0. 

We are using the default `1` here to signal that the user has set a value.